### PR TITLE
url-encoded basic auth to fetch avro schema in kafka 

### PR DESCRIPTION
**Pull request by dheerajathrey**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Kafka connector can fetch avro schema from schema registry with basic authentication using url-encoded credentials

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by evillique** [comment]
Is there a way to add a test for this? Maybe in `tests/integration/test_storage_kafka`?

File: src/Processors/Formats/Impl/AvroRowInputFormat.cpp [hunk 1]
```diff
@@ -734,6 +736,29 @@ class AvroConfluentRowInputFormat::SchemaRegistry
                 Poco::Net::HTTPRequest request(Poco::Net::HTTPRequest::HTTP_GET, url.getPathAndQuery(), Poco::Net::HTTPRequest::HTTP_1_1);
                 request.setHost(url.getHost());
 
+                if (!url.getUserInfo().empty()) {
```
Comment (evillique) [inline src/Processors/Formats/Impl/AvroRowInputFormat.cpp:739]:
```suggestion
                if (!url.getUserInfo().empty())
                {
```

File: src/Processors/Formats/Impl/AvroRowInputFormat.cpp [hunk 2]
```diff
@@ -734,6 +736,29 @@ class AvroConfluentRowInputFormat::SchemaRegistry
...
+                    Poco::Net::HTTPCredentials http_credentials;
+                    Poco::Net::HTTPBasicCredentials http_basic_credentials;
+                    std::string decoded_username;
+                    std::string decoded_password;
+
+                    http_credentials.fromUserInfo(url.getUserInfo());
+
+                    if (!http_credentials.getPassword().empty()) {
```
Comment (evillique) [inline src/Processors/Formats/Impl/AvroRowInputFormat.cpp:747]:
```suggestion
                    if (!http_credentials.getPassword().empty())
                    {
```

File: src/Processors/Formats/Impl/AvroRowInputFormat.cpp [hunk 3]
```diff
@@ -734,6 +736,29 @@ class AvroConfluentRowInputFormat::SchemaRegistry
...
+                        Poco::URI::decode(http_credentials.getUsername(), decoded_username);
+                        Poco::URI::decode(http_credentials.getPassword(), decoded_password);
+
+                        http_basic_credentials.setUsername(decoded_username);
+                        http_basic_credentials.setPassword(decoded_password);
+                    }
+                    else {
```
Comment (evillique) [inline src/Processors/Formats/Impl/AvroRowInputFormat.cpp:754]:
```suggestion
                    else
                    {
```

**Comment by kiliangrashoff** [comment]
Hi @dheerajathrey, are you planning to continue work on this PR? I'm trying to get a ClickHouse deployment to work with Aiven schema registry which uses basic auth, and it would really help me when this is merged.

**Comment by dheerajathrey** [comment]
@kiliangrashoff I have been busy with other work, I will continue work on this, this week.

**Comment by dheerajathrey** [comment]
@evillique I am facing problem when running integration test locally. Is there a way to trigger an integration test from the CI here?

**Comment by dheerajathrey** [comment]
@evillique  https://github.com/ClickHouse/ClickHouse/issues/40612
Do you happen to know if the method of running integration tests locally has changed?

**Comment by evillique** [comment]
> @evillique I am facing a problem with running integration tests locally. I explained the problem in detail in this issue. #40612 Do you happen to know if the method of running integration tests locally has changed?

I usually use `./runner` script as described in `tests/integration/README.md`

**Comment by kiliangrashoff** [comment]
Hi @dheerajathrey are you still working on this? If not could you explain where you got stuck, maybe we can find someone to help finish this PR.

**Comment by lesandie** [comment]
It would be nice to fix this ❤️ 

Some workaround but it is not very convenient as security is downgraded:

https://github.com/ClickHouse/ClickHouse/issues/32887#issuecomment-1427624770

**Comment by dheerajathrey** [comment]
Hi @kiliangrashoff I missed the email about the fix with PR [#41143](https://github.com/ClickHouse/ClickHouse/pull/41143) and the conversation on this current pr. I will work it next weekend.

**Comment by ilejn** [comment]
Hello @kiliangrashoff , could you help with https://github.com/ClickHouse/ClickHouse/pull/49664 review?

**Pull request closed without merging**